specifyweb/frontend/js_src/lib/components/QueryBuilder/FieldFilter.tsx

```diff
@@ -61,9 +68,234 @@ export const filtersWithDefaultValue = new Set<QueryFieldFilter>([
   'in',
 ]);
 
+const relativeDateItems = {
+  direction: [
+    { value: '+', title: queryText.future() },
+    {
+      value: '-',
+      title: queryText.past(),
+    },
+  ],
+  type: [
+    {
+      value: 'day',
+      title: queryText.day(),
+    },
+    { value: 'week', title: queryText.week() },
+    {
+      value: 'month',
+      title: queryText.month(),
+    },
+    {
+      value: 'year',
+      title: queryText.year(),
+    },
+  ],
+};
+
+function PickListSimple({
+  pickListItems,
+  value,
+  onBlur: handleBlur,
+  onChange: handleChange,
+}: {
+  readonly pickListItems: RA<PickListItemSimple>;
+  readonly value: string;
+  readonly onBlur: ({ target }: React.ChangeEvent<HTMLSelectElement>) => void;
+  readonly onChange: (value: string) => void;
+}): JSX.Element {
+  return (
+    <Select value={value} onBlur={handleBlur} onValueChange={handleChange}>
+      {pickListItems.map(({ title, value }) => (
+        <option key={value} value={value}>
+          {title}
+        </option>
+      ))}
+    </Select>
+  );
+}
+
+function extractValuesSimple<TYPE>(
+  target: HTMLInputElement | HTMLSelectElement
+): TYPE {
+  return target.value as unknown as TYPE;
+}
+
+function DateSplit({
```

Comment:
A good practice is to put parent component first, and then the components that are used in the parent (so `DateQueryInputField` should be defined before `DateSplit`).

This way when you read the code, you can just go from top to bottom rather than jump around a lot

Reply:
Thanks for refactoring this and other QueryBuilder files!